src/data/viteTrickFetcher.ts

```diff
@@ -0,0 +1,157 @@
+}
+
+
+function findDuplicateKeys(allTricks: z.infer<typeof DbTricksTableZod>[]) {
+
+  let idToTrickLookup: Record<number, (z.infer<typeof DbTricksTableZod>)[]> = createRecordLookup(allTricks);
+
+  return Object.entries(idToTrickLookup).filter(([_,e]) => e.length > 1).map(([k,v]) => ({
+    id: k,
+    technicalNames: v.map( e => e.technicalName)
+  }))
+
+}
+
+
+function createRecordLookup(allTricks: z.infer<typeof DbTricksTableZod>[]) {
+  let idToTrickLookup: Record<number, (z.infer<typeof DbTricksTableZod>)[]> = {};
+
+  allTricks.forEach(e => {
+    if (!idToTrickLookup[e.id]) {
+      idToTrickLookup[e.id] = [];
+    }
+      idToTrickLookup[e.id].push(e);
+    });
+  return idToTrickLookup;
+}
+
+/**
+ * Tricks contain two types of References: 
+ * - recommendedPrerequisites
+ * - variationOf
+ */
+function findUndefinedReferences(allTricks: z.infer<typeof DbTricksTableZod>[]) {
+  
+  let idToTrickLookup: Record<number, (z.infer<typeof DbTricksTableZod>)[]> = createRecordLookup(allTricks);
+
+  // issues define the ID of the Trick and an issue message
+  const issues: [number, string][] = []
+
+  allTricks.forEach( e => {
+    if(e.recommendedPrerequisites) {
+      e.recommendedPrerequisites.map( (f,i) => {
+        if(f[1] !== "official") {
+          issues.push([e.id, `All recommendedPrerequisites must limit themselves to official tricks (not the case for ${i+1}. Entry)`])
+        }
+        else if(!idToTrickLookup[f[0]]) {
+          issues.push([e.id, `${i+1}. Entry in recommendedPrerequisites references an ID that does not exist (id: ${f[0]}).`])
+        }
+        else if(f[0] == e.id) {
+          issues.push([e.id, `${i+1}. Entry in recommendedPrerequisites references itself (id: ${f[0]}).`])
+        }
+
+      })
+    }
+    if(e.variationOf) {
+      e.variationOf.map( (f,i) => {
+        if(f[1] !== "official") {
+          issues.push([e.id, `All variationOf-Entries must limit themselves to official tricks (not the case for ${i+1}. Entry)`])
+        }
+        else if(!idToTrickLookup[f[0]]) {
+          issues.push([e.id, `${i+1}. Entry in variationOf references an ID that does not exist (id: ${f[0]}).`])
+        }
+        else if(f[0] == e.id) {
+          issues.push([e.id, `${i+1}. Entry in variationOf references itself (id: ${f[0]}).`])
+        }
+      })
+    }
+  })
+
+  return issues;
+}
+
+
+
+export default async function viteGetAllTricks() {
+  // @ts-expect-error 
+  const pattern = join(fileURLToPath(import.meta.url), "..", "tricks", "*.yaml");
+  const allFilePaths = await globby(pattern)
+
+  const parsedYamlFiles = await Promise.allSettled(allFilePaths.map( e => fetchYamlFile(e)))
+  const erroredFiles = parsedYamlFiles.map( (e,i) => e.status === "rejected" ? ({error: e.reason, path: allFilePaths[i]}) : (undefined as never) ).filter( e => e);
+  const goodFiles = parsedYamlFiles.map( e => e.status === "fulfilled" ? e.value : (undefined as never)).filter( e => e)
+
+  if(erroredFiles.length) {
```

Comment:
Is it possible to use `Array.length` as a boolean expression to use in an `if` statement?

Reply:
It's possible, and this would work as intended, but `if(arr.length > 0)` would be more descriptive. But I don't remember intentionally keeping it like that. Will change :)

But yes.
`!!0` is false, `!!1` is true :D